File: backend/FleetMind.Api/Models/AuditLog.cs
using System;
using FleetMind.Api.Common;
using FleetMind.Api.Models.Common;

namespace FleetMind.Api.Models
{
    public class AuditLog : BaseEntity
    {
        public Guid? UserId { get; set; }
        public string UserName { get; set; } = null!;
        public string Action { get; set; } = null!;
        public string EntityName { get; set; } = null!;
        public string EntityId { get; set; } = null!;
        public string? Changes { get; set; }
        public DateTime Timestamp { get; set; }
    }
}

File: frontend/src/types/auditLog.ts
import { PaginationQuery } from './pagination';

export interface AuditLog {
  id: string;
  userId: string | null;
  userName: string;
  action: string;
  entityName: string;
  entityId: string;
  changes: string | null;
  timestamp: string;
}

export interface AuditLogQuery extends PaginationQuery {
  entityName?: string;
  action?: string;
  userId?: string;
  dateFrom?: string;
  dateTo?: string;
}
